src/Core__Option.res

```diff
@@ -111,3 +123,9 @@ let cmpU = (a, b, f) =>
   }
 
 let cmp = (a, b, f) => cmpU(a, b, (. x, y) => f(x, y))
+
+let zip = (a, b) =>
```

Comment:
I can definitely see this being useful. Would we maybe also want a zip3 etc.?

Reply:
What are some common use cases for this? We've opted to _not_ include `zip` style APIs in other modules because they seem a bit niche. Can revisit that decision of course though, but I'd like to understand more in what scenarios this is useful.

Reply:
Just checked around in other libraries for naming convention, since I wasn't entirely sure if it should be called `zip2` or `zip3`, but found that neither Rust or Scala has it. Though Scala interestingly has `unzip3`. But this lead me instead to conclude that there probably isn't much need for it.

Reply:
> What are some common use cases for this? We've opted to not include zip style APIs in other modules because they seem a bit niche. Can revisit that decision of course though, but I'd like to understand more in what scenarios this is useful.

Hmm, when I saw it I was thinking that there would have been a few places in our codebases where this could have come in handy, but when I actually checked now I didn't really find a compelling use case either. So yes, maybe better to leave it out for now. Do you have some compelling real-world usage example, @glennsl?

Reply:
Sorry, I missed your question before @zth. Mostly I think it's just to make some patterns cleaner and more readable. Here's a real world example:

```rescript
let maybeRatio = switch value->Option.zip(reference) {
| Some(0., _) | Some(_, 0.) | None => None
| Some(value, reference) => Some((value -. reference) /. reference)
}
```

Without `Option.zip` it would be:

```rescript
let maybeRatio = switch (value, reference) {
| (Some(0.) | None, _) | (_, Some(0.) | None) => None
| (Some(value), Some(reference)) => Some((value -. reference) /. reference)
}
```

Not a huge difference, but it does cut down on the noise a bit.


Another scenario I can think of is the possibility of making a `switch` into a one-liner:

```rescript
switch (a, b) {
| (Some(a), Some(b)) => a * b
| _ => 0
}
```

Could be written as:

```rescript
a->Option.zip(b)->Option.map(((a, b)) => a * b)
```

Though an `Option.map2` would probably serve this case better.

I'm absolutely fine with nominating this for a future `rescript-core-extras` library instead.

Reply:
And removed